src/sidebar/util/cfi.ts

```diff
@@ -0,0 +1,102 @@
+/**
+ * Compare two arrays.
+ *
+ * Arrays are compared as a sequence of values in priority order. If the two
+ * arrays are of different length but their common indexes have the same values,
+ * the shorter array is considered less than the longer one.
+ *
+ * This logic is similar to how eg. tuples are compared in Python.
+ */
+function compareArrays(
+  a: Array<number | string>,
+  b: Array<number | string>
+): number {
+  for (let i = 0; i < Math.min(a.length, b.length); i++) {
+    if (a[i] === b[i]) {
+      continue;
+    } else if (a[i] < b[i]) {
+      return -1;
+    } else if (a[i] > b[i]) {
+      return 1;
+    }
```

Comment:
Presumably the intended behavior is that if a `string` entry is compared to a `number` entry, the `string` is considered the larger. Should that be spelled out somewhere? I know it's an unexpected case...

Reply:
Good point. Comparing a number with a string results in coercing the string to a number. If that fails, the result is a `NaN` and comparisons involving `NaN` are always false, so what happened is that the current CFI step is ignored and comparison preceded with the next step. That's not actually terrible behavior considering that non-integer CFI steps are invalid. Nevertheless, it is an implementation accident that it works this way. I added explicit logic to make the ordering as you suggested (numbers are always less than strings).